crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -64,6 +82,24 @@ impl StateCallback {
     }
 }
 
+pub struct StateTransitionBuilder<'a, T>
```

Comment:
I love this builder pattern, however generally a builder pattern includes a final `.build()` function call. 
This feels more like a `StateTransitionConfig` or something to that extent. 
(Though tbf, this is something I don't feel too strongly about).